Abort non-interactive setup when no default is available to prevent
endless loop when stdin is not a tty.

diff -Naur --exclude=debian perl-5.8.8.orig/lib/CPAN/FirstTime.pm perl-5.8.8/lib/CPAN/FirstTime.pm
--- perl-5.8.8.orig/lib/CPAN/FirstTime.pm	2006-01-31 08:08:57.000000000 +1100
+++ perl-5.8.8/lib/CPAN/FirstTime.pm	2007-09-16 15:57:23.000000000 +1000
@@ -553,6 +553,9 @@
         $pos = 0 if $pos >= @$items;
 
         my $num = prompt($prompt,$default);
+        unless ($num or -t) {
+            die "No default for '$prompt',\naborting non-interactive setup.\n";
+        }
 
         @nums = split (' ', $num);
         my $i = scalar @$items;
